sycl/test-e2e/Basic/if_device.cpp

```diff
@@ -0,0 +1,45 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <sycl/sycl.hpp>
```

Comment:
Please, include minimal required set of headers: queue, buffer, accessor and `<sycl/ext/oneapi/experimental/if_device.hpp>`.

Reply:
I strongly disagree. The right process to do that is to update the spec/create extension, allocate resources to the activity, cleanup the headers and create automation/checks to automatically cleanup includes and only after that can we start requesting such changes in arbitrary PRs. Until then we have to treat end-to-end tests same way as examples and write them how customers would write their code according to the specification.